File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        left = 0
        right = sum(nums[1:])
        if nums[0]  == right + nums[0]:
            return 0
        for ptr in range(1,len(nums)):
            left += nums[ptr - 1]
            right -= nums[ptr]
            if left == right:
                return ptr
        return -1


        

File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    let right = nums.slice(1).reduce((acc, val) => acc + val, 0)
    let left = 0
    if(right +  nums[0] == nums[0]) { return 0}
    for(let i = 1; i< nums.length; i++){
        right -= nums[i - 1]
        left += nums[i]
        if(left == right){ return i }
    }
    return -1
};